From ad9dbda948355df298b73100a90df13f8d2b5baa Mon Sep 17 00:00:00 2001
From: Arjen <edi@minbzk.nl>
Date: Mon, 5 Feb 2024 13:16:43 +0100
Subject: [PATCH 5/5] Add environment to config

---
 app/dependency_injection/config.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/app/dependency_injection/config.py b/app/dependency_injection/config.py
index f70ca99..a0ce6e7 100644
--- a/app/dependency_injection/config.py
+++ b/app/dependency_injection/config.py
@@ -1,4 +1,6 @@
 import configparser
+from configparser import ExtendedInterpolation
+import os
 from typing import Any
 
 from app.models.swagger_config import SwaggerConfig
@@ -6,6 +8,11 @@ from app.models.swagger_config import SwaggerConfig
 _PATH = "max.conf"
 _CONFIG = None
 
+class ExtendedEnvInterpolation(configparser.ExtendedInterpolation):
+    """Interpolation which expands environment variables in values."""
+
+    def before_get(self, parser, section, option, value, defaults):
+        return os.path.expandvars(value)
 
 # pylint:disable=global-statement
 def get_config(path=None) -> configparser.ConfigParser:
@@ -18,7 +25,7 @@ def get_config(path=None) -> configparser.ConfigParser:
         path = _PATH
     if _CONFIG is None or _PATH != path:
         _PATH = path
-        _CONFIG = configparser.ConfigParser()
+        _CONFIG = configparser.ConfigParser(interpolation=ExtendedEnvInterpolation())
         _CONFIG.read(_PATH)
     return _CONFIG
 
-- 
2.43.0

